pkg/cloudprovider/provider/azure/create_delete_resources.go

```diff
@@ -0,0 +1,282 @@
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-04-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2018-04-01/network"
+	"github.com/Azure/go-autorest/autorest/to"
+	"github.com/golang/glog"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// deleteInterfacesByMachineUID will remove all network interfaces tagged with the specific machine's UID.
+// The machine has to be deleted or disassociated with the interfaces beforehand, since Azure won't allow
+// us to remove interfaces connected to a VM.
+func deleteInterfacesByMachineUID(ctx context.Context, c *config, machineUID types.UID) error {
```

Comment:
Can you quickly explain why we need this? The terraform azure provider does not do it: https://github.com/terraform-providers/terraform-provider-azurerm/blob/aab882f476253e05bae288062c42421d7bf77746/azurerm/resource_arm_virtual_machine.go#L747

Reply:
https://github.com/terraform-providers/terraform-provider-azurerm/blob/aab882f476253e05bae288062c42421d7bf77746/azurerm/resource_arm_network_interface.go#L394